[Doc] Document Filter API (#837)
elastic/apm-agent-dotnet: docs/public-api.asciidoc

NOTE: Return value of <<convenient-capture-span>> method overloads that accept Task (or Task<T>) is the same Task (or Task<T>) instance as the one passed as the argument so if your application should continue only after the task completes you have to call <<convenient-capture-span>> with `await` keyword.

NOTE: Code samples above use `Elastic.Apm.Agent.Tracer.CurrentTransaction`. In production code you should make sure the `CurrentTransaction` is not `null`.

[float]
[[filter-api]]
== Filter API (added[1.5])

Use `Agent.AddFilter(filter)` to supply a filter callback.

> Each filter callback will be called just before data is being sent to the APM Server. This will allow you to manipulate the data being sent, for instance to remove sensitive information like passwords etc.

Reviewer: ```suggestion
Each filter callback will be called just before data is sent to the APM Server. This allows you to manipulate the data being sent, like to remove sensitive information such as passwords.
```

> Each filter callback will be called in the order they were added, and will receive a payload object as the only argument, containing the data about to be sent to the APM Server.

Reviewer: ```suggestion
Each filter callback is called in the order they are added and will receive a payload object containing the data about to be sent to the APM Server as the only argument.
```

The filter callback is synchronous and should return the manipulated payload object. If a filter callback doesn’t return any value or returns a falsy value, the remaining filter callback will not be called and the payload will not be sent to the APM Server.

There are 3 overloads of the `Agent.AddFilter` method with the following arguments:

- `Func<ITransaction, ITransaction>`: A filter called for every transaction
- `Func<ISpan, ISpan>`: A filter called for every span

> - `Func<IError, IError>`: A filter called for every error.

Reviewer: Inconsistent period usage here. I recommend adding `.` to all.

> 

Author: Maybe we could add something here, like 

> Below are some example usages of the `Agent.AddFilter` method. 

Not sure what's the best way to do that.

Reviewer: Your text sounds good to me! Although I might saw two words: `Below are some usage examples of the Agent.AddFilter method.`
